libs/sdk-ui-ext/src/insightView/InsightView.tsx

```diff
@@ -163,8 +167,8 @@ class InsightViewCore extends React.Component<IInsightViewProps & WrappedCompone
         if (!insight || isEqual(insight, this.state.insight)) {
             return;
         }
-
-        this.setState({ insight });
+        this.props.onInsightLoaded?.(insight);
+        this.setState({ insight, insightTitle: this.resolveInsightTitle(insight) });
```

Comment:
It is not necessary to store the title in the state, right? We ca  always call the `resolveInsightTitle` whenever we want the value (the function is not heavy so there is no need to "memoize" it). And it is usually better to compute the derived data from state, not store them there.

Reply:
Resolved. It seemed better to me that the title is calculated just once after insight is loaded. It is true tho that the solution without saving the title into the state has better readability.